src/app/state/assets/selectors/asset.selectors.ts

```diff
@@ -0,0 +1,28 @@
+import {createSelector} from '@ngrx/store';
+import {DateRange} from '../../../shared/models/date-range';
+import {formFeature} from '../../form/reducers/form.reducer';
+import {assetFeature} from '../reducers/asset.reducer';
+import type {DataInterval} from '../../../shared/models/interval';
+import type {TimeRange} from '../../../shared/models/time-range';
+
+export const selectAvailableDataInterval = createSelector(assetFeature.selectAssets, (assets): DataInterval[] => {
+  return [...new Set(assets?.map((asset) => asset.interval))];
```

Comment:
To make it more explicit if `assets` is `undefined`
```suggestion
  return assets ? [...new Set(assets.map((asset) => asset.interval))] : [];
```